cmd/kube-apiserver/app/server.go

```diff
@@ -643,15 +643,15 @@ func BuildAuthorizer(s *options.ServerRunOptions, versionedInformers clientgoinf
 	return authorizationConfig.New()
 }
 
-// completedServerRunOptions is a private wrapper that enforces a call of Complete() before Run can be invoked.
-type completedServerRunOptions struct {
+// CompletedServerRunOptions is a private wrapper that enforces a call of Complete() before Run can be invoked.
```

Comment:
i think it is intentional to use completedServerRunOptions as it is a private wrapper

Reply:
I see, Then it should not be returned... I'll see what and why Compute does this, and maybe, figure out what to do about the return.

Reply:
So far, I think this type is private to keep it as some kind of Singleton, preventing users from modifying it once created.
I've found #60483, when this was introduced, and @hzxuzhonghu and @sttts discuss a little bit about this.
I've also found golang/lint#210 on this subject.
If I am understanding it all right, the best way to address this is to create a public interface for the return object, so it gets included in godoc and documented, while still not being modifiable. Maybe this interface should provide methods to get the fields from the underlying struct. Is that right?
What I could not find is the usage of this returned object in other places in the system, so I'm fearing to break something critical and/or getting a lot of changes to be done.

Reply:
This is private intentionally. It's sad that golint does not like that pattern. But the solution is not to remove it. It has had value numerous times to not be able to create such an object without calling `Complete`. And an interface is not a solution either other than making golint happy: the  https://github.com/golang/lint/issues/210 is about a different use-case, namely to hide implementations. Here it's about forbidding instantiation, but being able to access fields. That's a different issue.

The solution is to fix golint to allow local exceptions of rules. Sometimes rules do not apply.